plugins/org.python.pydev.parser/tests/org/python/pydev/parser/FStringsParserTest.java

```diff
@@ -76,7 +75,24 @@ private void checkError(String str, String... expected) throws ParseException {
         }
     }
 
+    private Tuple<FStringsAST, List> checkStr(String str, Set<String> exprs)
+            throws ParseException, BadLocationException {
+        Tuple<FStringsAST, List> ret = checkNoError(str);
+        IDocument doc = new Document(str);
+        // ret.o1.dump(doc);
+        Set<String> found = new HashSet<>();
+        for (SimpleNode b : ret.o1.getBalancedExpressionsToBeEvaluatedInRegularGrammar()) {
+            String contents = b.getContentsFromString(doc);
+            found.add(contents);
+        }
+
+        assertEquals(exprs.toString(), found.toString());
+        return ret;
+    }
+
     public void testFStringParsing() throws ParseException, BadLocationException {
+        checkStr("\\N{foo}", ArrayUtils.asSet(""));
```

Comment:
Please add some cases testing the string replacement directly (as mentioned before) and add more tests for the corner cases listed before.

Also, you should be able to call `checkExprs("\\N{foo}", ArrayUtils.asSet());` and it should work (you don't need the `checkStr` -- I think you have become confused because doing `ArrayUtils.asSet()` and `ArrayUtils.asSet("")`, both have the same representation in the error output, even though `assertEquals(ArrayUtils.asSet(), ArrayUtils.asSet(""));` is really expected to fail (because one is really an empty set and the other is a set with a single entry but which doesn't have any representation.